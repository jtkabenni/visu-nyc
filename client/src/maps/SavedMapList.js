import React from "react";
import SavedMapCard from "./SavedMapCard";

function SavedMapList({ maps }) {
  return (
    <div className="map-list">
      <h3>All maps</h3>
      {maps.length ? (
        <>
          {maps.map((m) => (
            <SavedMapCard key={m.id} m={m}></SavedMapCard>
          ))}
        </>
      ) : (
        <i>No maps added yet.</i>
      )}
    </div>
  );
}

export default SavedMapList;
