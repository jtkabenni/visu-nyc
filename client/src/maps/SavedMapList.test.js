import React from "react";
import { render } from "@testing-library/react";
import SavedMapList from "./SavedMapList";

let maps = [
  {
    name: "BK maps",
    use_username: "jennytak",
    file: "",
    note: "bk map note",
  },
  {
    name: "BK maps 2",
    use_username: "jennytak",
    file: "",
    note: "bk map note",
  },
];
it("renders without crashing", function () {
  render(<SavedMapList maps={maps} />);
});

it("matches snapshot with no jobs", function () {
  const { asFragment } = render(<SavedMapList maps={maps} />);
  expect(asFragment()).toMatchSnapshot();
});
