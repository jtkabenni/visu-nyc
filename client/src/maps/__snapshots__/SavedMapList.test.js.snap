// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with no jobs 1`] = `
<DocumentFragment>
  <div
    class="map-list"
  >
    <h3>
      All maps
    </h3>
    <div
      class="map-card"
    >
      <p>
        <b>
          BK maps 
        </b>
      </p>
      <p>
        bk map note
      </p>
    </div>
    <div
      class="map-card"
    >
      <p>
        <b>
          BK maps 2 
        </b>
      </p>
      <p>
        bk map note
      </p>
    </div>
  </div>
</DocumentFragment>
`;
